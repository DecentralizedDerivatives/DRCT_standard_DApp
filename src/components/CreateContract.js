import React, { Component } from 'react';
import { connect } from 'react-redux';
import PropTypes from 'prop-types';
import CreateContractFormContainer from './CreateContractFormContainer';
import SendFundsFormContainer from './SendFundsFormContainer';
import { sendSendFundsOrder } from '../actions/orderActions';

export class CreateContract extends Component {
  constructor(props) {
    super(props);
    this.state = {
      sendFundsOpen: false,
      resultsMessage: ''
    };
  }

  componentWillReceiveProps(nextProps) {
    if (nextProps.newContractCreateError) {
      this.setState({
        resultsMessage: `Error: ${this.props.newContractError}`,
        sendFundsOpen: false
      });
    } else if (nextProps.newContract.address) {
      this.setState({
        resultsMessage: `Address result ${this.props.newContract.address}`,
        sendFundsOpen: true
      });
    } else if (nextProps.newContractFundsError) {
      this.setState({
        resultsMessage: `Error: ${this.props.newContractFundsError}`,
        sendFundsOpen: false
      });
    } else if (nextProps.newContract.funded) {
      this.setState({
        resultsMessage: `Contract successfully funded.`,
        sendFundsOpen: false
      });
    }
  }

  handleSkipCreate = e => {
    e.preventDefault();
    this.setState({sendFundsOpen: true});
  }
  handleSendFundsDirectly = (details) => {
    // console.log('FUNDING DETAILS', details);
    this.props.sendSendFundsOrder(details, this.props.userAccount);
  }
  handleSendFundsClick = e => {
    // console.log('CONTRACT DETAILS', this.props.newContract);
    this.props.sendSendFundsOrder(this.props.newContract, this.props.userAccount);
  };

  render() {
    return(
      this.state.sendFundsOpen ? (
        <div className="create-contract" >
          <div className="modal-background" onClick={this.props.close}></div>
          <div className="modal">
            {this.props.newContract.address ?
              (
                <div id="send-funds">
                  <h3 className="created-address">
                    Address Result : {this.props.newContract.address}
                  </h3>
                  {this.props.newContractFundsError}
                  <button onClick={this.handleSendFundsClick}>Send Funds</button>
                </div>
              ) : (
                <SendFundsFormContainer
                  sendFunds={this.handleSendFundsDirectly} />
              )
            }
          </div>
        </div>
      ):(
        <div className="create-contract">
        <div className="modal-background" onClick={this.props.close}></div>
        <div className="modal">
          <CreateContractFormContainer
            name="createContractForm"
            handleSkipCreate={this.handleSkipCreate} />
        </div>
      </div>
      )
    )
  }
}

CreateContract.propTypes = {
  sendSendFundsOrder: PropTypes.func.isRequired,
  userAccount: PropTypes.string,
  newContract: PropTypes.object,
  newContractFundsError: PropTypes.string,
  createContractAmount: PropTypes.number
};


const mapStateToProps = state => ({
  userAccount: state.user.userAccount,
  newContract: state.contract.newContract,
  newContractCreateError: state.contract.newContractCreateError,
  newContractFundsError: state.contract.newContractFundsError
});

export default connect(
  mapStateToProps,
  { sendSendFundsOrder }
)(CreateContract);
