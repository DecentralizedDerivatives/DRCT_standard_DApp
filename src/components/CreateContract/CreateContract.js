import React, {Component} from 'react';
import PropTypes from 'prop-types';
import withStyles from 'material-ui/styles/withStyles';
import Button from 'material-ui/Button';
import TextField from 'material-ui/TextField';
import Grid from 'material-ui/Grid';
import Typography from 'material-ui/Typography';
import Dialog, {DialogContent} from 'material-ui/Dialog';
import {DatePicker} from 'material-ui-pickers';
import {CircularProgress} from 'material-ui/Progress';
import styles from './styles';
import Dropdown from '../Dropdown';
import {Factory, UserContract, web3} from '../../ethereum';

class CreateContract extends Component {
  static propTypes = {
    classes: PropTypes.object.isRequired,
    open: PropTypes.bool.isRequired,
    toggle: PropTypes.func.isRequired,
  };

  static durations = ['One weeks', 'Two weeks'];
  static currency = ['BTC/USD', 'ETH/USD'];

  state = {
    open: false,
    duration: '',
    currency: '',
    amount: 0.1,
    contractAddress: '',
    txId: '',
    selectedDate: new Date(),
    loading: false,
    disabled: false,
    created: false,
  };

  handleChange = event => {
    this.setState({[event.target.name]: event.target.value});
  };

  handleDateChange = date => {
    this.setState({selectedDate: date});
  };

  handleTextfieldChange = name => event => {
    this.setState({
      [name]: event.target.value,
    });
  };

  createContract = async () => {
    const factory = await Factory.deployed();
    const accounts = await web3.eth.getAccounts();

    let date = Number(
      (new Date(this.state.selectedDate).getTime() / 1000).toFixed(0)
    );

    date = date - date % 86400;

    let response, error;

    this.setState({loading: true, disabled: true, showAddress: true});

    try {
      response = await factory.deployContract(date, {
        from: accounts[0],
        gas: 4000000,
      });
    } catch (err) {
      error = err;
    }

    this.setState({loading: false});

    if (error) {
      // Add error handling
      this.setState({txId: error.tx, error: true, disabled: false});
      return;
    }

    this.setState({
      showSendFunds: true,
      txId: response.tx,
      contractAddress: response.logs[0].args._created,
    });
  };

  sendFunds = async () => {
    const userContract = await UserContract.deployed();
    const accounts = await web3.eth.getAccounts();
    console.log(this.state.contractAddress);
    console.log(accounts[0]);

    let _value = 1e18 * this.state.amount;
    console.log(this.state.contractAddress,_value,_value*2);
    let response, error;
    userContract.Initiate(this.state.contractAddress,_value,{
        from: accounts[0],
        gas: 4000000,
        value: _value*2
      }).then((res,err) =>{
        if(err){
          console.log('Error Message:', err);
        }
        else(console.log('Succesful Initiation of Contract!: ',res));
      })
    this.props.toggle
  };

  render() {
    const {classes} = this.props;

    return (
      <div>
        <Dialog
          open={this.props.open}
          onClose={this.props.toggle}
          PaperProps={{className: classes.paper}}
        >
          <DialogContent className={classes.dialogContent}>
            <div className={classes.inputContainer}>
              <Typography className={classes.title}>Contract Type</Typography>
              <Grid container justify="space-between">
                <Grid item>
                  <Dropdown
                    menuItems={CreateContract.durations}
                    value={this.state.duration}
                    name="duration"
                    onChange={this.handleChange}
                    className={classes.duration}
                  />
                </Grid>
                <Grid item>
                  <Dropdown
                    menuItems={CreateContract.currency}
                    value={this.state.currency}
                    name="currency"
                    onChange={this.handleChange}
                    className={classes.currency}
                  />
                </Grid>
              </Grid>
            </div>

            <div className={classes.inputContainer}>
              <Typography className={classes.title}>Start Date</Typography>

              <DatePicker
                value={this.state.selectedDate}
                onChange={this.handleDateChange}
                animateYearScrolling={false}
                className={classes.fullWidth}
                format={'MMMM D YYYY'}
              />
            </div>

            <div className={classes.inputContainer}>
              <Typography className={classes.title}>Amount of Ether</Typography>

              <TextField
                id="amount"
                value={Number(this.state.amount)}
                type="number"
                onChange={this.handleTextfieldChange('amount')}
                className={classes.fullWidth}
                helperText="Must be at least 0.1"
              />
            </div>
            <Button
              className={
                this.state.disabled ? classes.buttonDisabled : classes.button
              }
              disabled={this.state.disabled}
              onClick={this.createContract}
            >
              <Typography className={classes.buttonText}>
                Create Contract
              </Typography>
            </Button>
          </DialogContent>

          {this.state.showAddress && <div className={classes.line} />}
          {this.state.showAddress && (
            <DialogContent className={classes.addressResultContainer}>
              <div className={classes.inputContainer}>
                <Grid
                  container
                  direction="row"
                  alignItems="stretch"
                  justify="space-between"
                >
                  <Grid item>
                    <Typography className={classes.title}>
                      Address Result
                    </Typography>
                  </Grid>

                  <Grid item>
                    {this.state.loading && (
                      <Grid container direction="row" alignItems="stretch">
                        <Grid item>
                          <Typography className={classes.waiting}>
                            Waiting for confirmation...
                          </Typography>
                        </Grid>

                        <Grid item>
                          <CircularProgress
                            className={classes.progress}
                            size={12}
                            thickness={5}
                          />
                        </Grid>
                      </Grid>
                    )}
                  </Grid>
                </Grid>

                {this.state.contractAddress && (
                  <Typography className={classes.contractAddress}>
                    {this.state.contractAddress}
                  </Typography>
                )}
              </div>
            </DialogContent>
          )}

          {this.state.showSendFunds && <div className={classes.line} />}
          {this.state.showSendFunds && (
            <DialogContent className={classes.sendFundsContainer}>
              <Button className={classes.button} onClick={this.sendFunds}>
                <Typography className={classes.buttonText}>
                  Send Funds
                </Typography>
              </Button>
            </DialogContent>
          )}
        </Dialog>
      </div>
    );
  }
}

export default withStyles(styles)(CreateContract);
