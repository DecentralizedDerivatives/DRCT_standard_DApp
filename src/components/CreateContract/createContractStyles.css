.input-container {
  margin-bottom: 15px;
}

.input-title {
  font-weight: 500;
  font-size: 13px;
}

.button-text {
  color: #fff;
}

.button {
  width: 100%;
  background-color: #114770;
}

.button:hover {
  background-color: #9fb5c5;
}

.button-disabled: {
  width: 100%;
  background-color: #9fb5c5;
}

.contract-address {
  font-weight: 500;
  font-size: 13px;
}

.waiting {
  font-weight: 300;
  font-size: 13px;
  color: rgba(0, 0, 0, 0.54);
  position: relative;
  float: right;
}
