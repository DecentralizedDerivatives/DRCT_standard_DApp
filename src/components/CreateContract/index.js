import CreateContract from './CreateContract';

export default CreateContract;
