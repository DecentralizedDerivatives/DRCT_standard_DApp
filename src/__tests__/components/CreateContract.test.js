import CreateContract from '../../components/CreateContract';

jest.mock('../../actions/orderActions');

function setup(overrides) {
  const store = initStore();

  const close = jest.fn();
  const props = { store, close, ...overrides };

  const wrapper = shallow(<CreateContract {...props} />).dive();

  return {
    wrapper,
    close,
  };
}

describe('<CreateContract />', () => {
  it('renders the component', () => {
    const { wrapper } = setup();
    expect(wrapper).toMatchSnapshot();
  });

  it('renders contract create error', () => {
    const { wrapper } = setup();
    wrapper.setProps({ newContractCreateError: 'Error hapenned.' });
    expect(wrapper).toMatchSnapshot();
  });

  it('renders funding error', () => {
    const { wrapper } = setup();
    wrapper.setProps({ newContractFundsError: 'Error hapenned.' });
    expect(wrapper).toMatchSnapshot();
  });

  it('renders funding', () => {
    const { wrapper, close } = setup();
    wrapper.setProps({ newContract: { funded: true } });
    expect(close).toBeCalledTimes(1);
  });

  it('handles empty setProps', () => {
    const { wrapper } = setup();
    wrapper.setProps({});
  });

  it('renders new contract', () => {
    const { wrapper } = setup();
    wrapper.setProps({ newContract: { address: '0x000...' } });
    expect(wrapper).toMatchSnapshot();
  });
});
