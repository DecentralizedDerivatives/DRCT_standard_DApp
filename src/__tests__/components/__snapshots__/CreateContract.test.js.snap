// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<CreateContract /> renders contract create error 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <CreateContract
    close={[MockFunction]}
    dispatch={[Function]}
    newContract={
      Object {
        "address": "",
        "amount": 0,
        "currency": "",
        "duration": "",
        "funded": false,
        "fundedTx": "",
        "id": "",
        "startDate": "",
      }
    }
    newContractCreateError="Error hapenned."
    newContractFundsError={null}
    store={
      Object {
        "dispatch": [Function],
        "getState": [Function],
        "replaceReducer": [Function],
        "subscribe": [Function],
        Symbol(observable): [Function],
      }
    }
    storeSubscription={
      Subscription {
        "listeners": Object {
          "clear": [Function],
          "get": [Function],
          "notify": [Function],
          "subscribe": [Function],
        },
        "onStateChange": [Function],
        "parentSub": undefined,
        "store": Object {
          "dispatch": [Function],
          "getState": [Function],
          "replaceReducer": [Function],
          "subscribe": [Function],
          Symbol(observable): [Function],
        },
        "unsubscribe": [Function],
      }
    }
    userAccount=""
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <div
          className="modal-background"
          onClick={[MockFunction]}
        />,
        <div
          className="modal"
        >
          <Connect(ReduxForm)
            name="createContractForm"
          />
        </div>,
      ],
      "className": "create-contract",
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "className": "modal-background",
          "onClick": [MockFunction],
        },
        "ref": null,
        "rendered": null,
        "type": "div",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": <Connect(ReduxForm)
            name="createContractForm"
          />,
          "className": "modal",
        },
        "ref": null,
        "rendered": Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "name": "createContractForm",
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        "type": "div",
      },
    ],
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <div
            className="modal-background"
            onClick={[MockFunction]}
          />,
          <div
            className="modal"
          >
            <Connect(ReduxForm)
              name="createContractForm"
            />
          </div>,
        ],
        "className": "create-contract",
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "className": "modal-background",
            "onClick": [MockFunction],
          },
          "ref": null,
          "rendered": null,
          "type": "div",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": <Connect(ReduxForm)
              name="createContractForm"
            />,
            "className": "modal",
          },
          "ref": null,
          "rendered": Object {
            "instance": null,
            "key": undefined,
            "nodeType": "class",
            "props": Object {
              "name": "createContractForm",
            },
            "ref": null,
            "rendered": null,
            "type": [Function],
          },
          "type": "div",
        },
      ],
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getDerivedStateFromProps": true,
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
  },
}
`;

exports[`<CreateContract /> renders funding error 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <CreateContract
    close={
      [MockFunction] {
        "calls": Array [
          Array [],
        ],
        "results": Array [
          Object {
            "isThrow": false,
            "value": undefined,
          },
        ],
      }
    }
    dispatch={[Function]}
    newContract={
      Object {
        "address": "",
        "amount": 0,
        "currency": "",
        "duration": "",
        "funded": false,
        "fundedTx": "",
        "id": "",
        "startDate": "",
      }
    }
    newContractCreateError={null}
    newContractFundsError="Error hapenned."
    store={
      Object {
        "dispatch": [Function],
        "getState": [Function],
        "replaceReducer": [Function],
        "subscribe": [Function],
        Symbol(observable): [Function],
      }
    }
    storeSubscription={
      Subscription {
        "listeners": Object {
          "clear": [Function],
          "get": [Function],
          "notify": [Function],
          "subscribe": [Function],
        },
        "onStateChange": [Function],
        "parentSub": undefined,
        "store": Object {
          "dispatch": [Function],
          "getState": [Function],
          "replaceReducer": [Function],
          "subscribe": [Function],
          Symbol(observable): [Function],
        },
        "unsubscribe": [Function],
      }
    }
    userAccount=""
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <div
          className="modal-background"
          onClick={
            [MockFunction] {
              "calls": Array [
                Array [],
              ],
              "results": Array [
                Object {
                  "isThrow": false,
                  "value": undefined,
                },
              ],
            }
          }
        />,
        <div
          className="modal"
        >
          <Connect(ReduxForm)
            name="createContractForm"
          />
        </div>,
      ],
      "className": "create-contract",
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "className": "modal-background",
          "onClick": [MockFunction] {
            "calls": Array [
              Array [],
            ],
            "results": Array [
              Object {
                "isThrow": false,
                "value": undefined,
              },
            ],
          },
        },
        "ref": null,
        "rendered": null,
        "type": "div",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": <Connect(ReduxForm)
            name="createContractForm"
          />,
          "className": "modal",
        },
        "ref": null,
        "rendered": Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "name": "createContractForm",
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        "type": "div",
      },
    ],
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <div
            className="modal-background"
            onClick={
              [MockFunction] {
                "calls": Array [
                  Array [],
                ],
                "results": Array [
                  Object {
                    "isThrow": false,
                    "value": undefined,
                  },
                ],
              }
            }
          />,
          <div
            className="modal"
          >
            <Connect(ReduxForm)
              name="createContractForm"
            />
          </div>,
        ],
        "className": "create-contract",
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "className": "modal-background",
            "onClick": [MockFunction] {
              "calls": Array [
                Array [],
              ],
              "results": Array [
                Object {
                  "isThrow": false,
                  "value": undefined,
                },
              ],
            },
          },
          "ref": null,
          "rendered": null,
          "type": "div",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": <Connect(ReduxForm)
              name="createContractForm"
            />,
            "className": "modal",
          },
          "ref": null,
          "rendered": Object {
            "instance": null,
            "key": undefined,
            "nodeType": "class",
            "props": Object {
              "name": "createContractForm",
            },
            "ref": null,
            "rendered": null,
            "type": [Function],
          },
          "type": "div",
        },
      ],
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getDerivedStateFromProps": true,
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
  },
}
`;

exports[`<CreateContract /> renders new contract 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <CreateContract
    close={
      [MockFunction] {
        "calls": Array [
          Array [],
        ],
        "results": Array [
          Object {
            "isThrow": false,
            "value": undefined,
          },
        ],
      }
    }
    dispatch={[Function]}
    newContract={
      Object {
        "address": "0x000...",
      }
    }
    newContractCreateError={null}
    newContractFundsError={null}
    store={
      Object {
        "dispatch": [Function],
        "getState": [Function],
        "replaceReducer": [Function],
        "subscribe": [Function],
        Symbol(observable): [Function],
      }
    }
    storeSubscription={
      Subscription {
        "listeners": Object {
          "clear": [Function],
          "get": [Function],
          "notify": [Function],
          "subscribe": [Function],
        },
        "onStateChange": [Function],
        "parentSub": undefined,
        "store": Object {
          "dispatch": [Function],
          "getState": [Function],
          "replaceReducer": [Function],
          "subscribe": [Function],
          Symbol(observable): [Function],
        },
        "unsubscribe": [Function],
      }
    }
    userAccount=""
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <div
          className="modal-background"
          onClick={
            [MockFunction] {
              "calls": Array [
                Array [],
              ],
              "results": Array [
                Object {
                  "isThrow": false,
                  "value": undefined,
                },
              ],
            }
          }
        />,
        <div
          className="modal"
        >
          <Connect(ReduxForm)
            name="createContractForm"
          />
        </div>,
      ],
      "className": "create-contract",
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "className": "modal-background",
          "onClick": [MockFunction] {
            "calls": Array [
              Array [],
            ],
            "results": Array [
              Object {
                "isThrow": false,
                "value": undefined,
              },
            ],
          },
        },
        "ref": null,
        "rendered": null,
        "type": "div",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": <Connect(ReduxForm)
            name="createContractForm"
          />,
          "className": "modal",
        },
        "ref": null,
        "rendered": Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "name": "createContractForm",
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        "type": "div",
      },
    ],
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <div
            className="modal-background"
            onClick={
              [MockFunction] {
                "calls": Array [
                  Array [],
                ],
                "results": Array [
                  Object {
                    "isThrow": false,
                    "value": undefined,
                  },
                ],
              }
            }
          />,
          <div
            className="modal"
          >
            <Connect(ReduxForm)
              name="createContractForm"
            />
          </div>,
        ],
        "className": "create-contract",
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "className": "modal-background",
            "onClick": [MockFunction] {
              "calls": Array [
                Array [],
              ],
              "results": Array [
                Object {
                  "isThrow": false,
                  "value": undefined,
                },
              ],
            },
          },
          "ref": null,
          "rendered": null,
          "type": "div",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": <Connect(ReduxForm)
              name="createContractForm"
            />,
            "className": "modal",
          },
          "ref": null,
          "rendered": Object {
            "instance": null,
            "key": undefined,
            "nodeType": "class",
            "props": Object {
              "name": "createContractForm",
            },
            "ref": null,
            "rendered": null,
            "type": [Function],
          },
          "type": "div",
        },
      ],
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getDerivedStateFromProps": true,
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
  },
}
`;

exports[`<CreateContract /> renders the component 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <CreateContract
    close={[MockFunction]}
    dispatch={[Function]}
    newContract={
      Object {
        "address": "",
        "amount": 0,
        "currency": "",
        "duration": "",
        "funded": false,
        "fundedTx": "",
        "id": "",
        "startDate": "",
      }
    }
    newContractCreateError={null}
    newContractFundsError={null}
    store={
      Object {
        "dispatch": [Function],
        "getState": [Function],
        "replaceReducer": [Function],
        "subscribe": [Function],
        Symbol(observable): [Function],
      }
    }
    storeSubscription={
      Subscription {
        "listeners": Object {
          "clear": [Function],
          "get": [Function],
          "notify": [Function],
          "subscribe": [Function],
        },
        "onStateChange": [Function],
        "parentSub": undefined,
        "store": Object {
          "dispatch": [Function],
          "getState": [Function],
          "replaceReducer": [Function],
          "subscribe": [Function],
          Symbol(observable): [Function],
        },
        "unsubscribe": [Function],
      }
    }
    userAccount=""
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <div
          className="modal-background"
          onClick={[MockFunction]}
        />,
        <div
          className="modal"
        >
          <Connect(ReduxForm)
            name="createContractForm"
          />
        </div>,
      ],
      "className": "create-contract",
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "className": "modal-background",
          "onClick": [MockFunction],
        },
        "ref": null,
        "rendered": null,
        "type": "div",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": <Connect(ReduxForm)
            name="createContractForm"
          />,
          "className": "modal",
        },
        "ref": null,
        "rendered": Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "name": "createContractForm",
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        "type": "div",
      },
    ],
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <div
            className="modal-background"
            onClick={[MockFunction]}
          />,
          <div
            className="modal"
          >
            <Connect(ReduxForm)
              name="createContractForm"
            />
          </div>,
        ],
        "className": "create-contract",
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "className": "modal-background",
            "onClick": [MockFunction],
          },
          "ref": null,
          "rendered": null,
          "type": "div",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": <Connect(ReduxForm)
              name="createContractForm"
            />,
            "className": "modal",
          },
          "ref": null,
          "rendered": Object {
            "instance": null,
            "key": undefined,
            "nodeType": "class",
            "props": Object {
              "name": "createContractForm",
            },
            "ref": null,
            "rendered": null,
            "type": [Function],
          },
          "type": "div",
        },
      ],
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getDerivedStateFromProps": true,
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
  },
}
`;
